apps/ledger-live-mobile/android/app/src/main/java/com/ledger/live/LocationHelperModule.kt

```diff
@@ -0,0 +1,224 @@
+ * - FusedLocationProviderClient which provides a way to listen to the location service state updates
+ */
+class LocationHelperModule(private val reactContext: ReactApplicationContext) :
+    ReactContextBaseJavaModule(reactContext) {
+
+    // Resolves with a success or error value. reject is not used because we want to pass an
+    // information to the JS code.
+    private var locationPromise: Promise? = null
+
+    // LocationRequest used by both LocationSettings and FusedLocationProviderClient
+    private var locationRequest: LocationRequest = LocationRequest.create();
+
+    private var fusedLocationProviderClient: FusedLocationProviderClient;
+
+    // Listens to result from the location enable request activity.
+    // Request sent from requestEnableLocation.
+    private val activityEventListener = object : BaseActivityEventListener() {
+        override fun onActivityResult(
+            activity: Activity?,
+            requestCode: Int,
+            resultCode: Int,
+            data: Intent?
+        ) {
+            if (requestCode === REQUEST_ENABLE_LOCATION) {
+                when (resultCode) {
+                    Activity.RESULT_OK -> {
+                        locationPromise?.resolve(SUCCESS_LOCATION_ENABLED);
+                    }
+                    Activity.RESULT_CANCELED -> {
+                        locationPromise?.resolve(ERROR_USER_DENIED_LOCATION);
+                    }
+                }
+                locationPromise = null;
+                return
+            }
+            super.onActivityResult(activity, requestCode, resultCode, data)
+        }
+    }
+
+    // Handler to listen to location services updates (turned on/off)
+    private val locationServicesUpdateCallback = object : LocationCallback() {
+        override fun onLocationAvailability(p0: LocationAvailability) {
+            val isLocationAvailable = p0.isLocationAvailable
+
+            if (isLocationAvailable) {
+                reactContext
+                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+                    .emit("LocationServiceUpdated", Arguments.createMap().apply {
+                        putString("service", "enabled")
+                    })
+            } else {
+                reactContext
+                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+                    .emit("LocationServiceUpdated", Arguments.createMap().apply {
+                        putString("service", "disabled")
+                    })
+            }
+            super.onLocationAvailability(p0)
+        }
+    }
+
+    init {
+        // PRIORITY_HIGH_ACCURACY:  request the most precise location possible.
+        // With this setting, the location services are more likely to use GPS to determine the location.
+        // The given permissions will also influence the level of precision.
+        locationRequest.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
```

Comment:
I get this deprecation warning for `PRIORITY_HIGH_ACCURACY`:
```
This constant is deprecated. Use Priority.PRIORITY_HIGH_ACCURACY instead.
```

Reply:
Added comment on why i had to use this constant